### ISSUE TITLE
ES6+ features using Babel?

### ISSUE THREAD

**Issue by ekmartin** [l12s/node-slack-client#18]
Since this is a complete rewrite of the code base, maybe it would be a good idea to use something like [Babel](https://babeljs.io/) to make use of ES6+ features? Since this is a pretty object oriented codebase stuff like ES6 classes and extends/super would probably fit quite well (in addition to features like imports/exports etc.).

Of course this would require a build step for development, which would make the startup time of the tests increase. However, users of the library wouldn't notice any difference, since you can use an npm prepublish step to build the code, and put anything else than the built code into `.npmignore`.

You've probably already had this discussion before, so if that's the case I apologize. Would to love to hear some of the arguments though.

If it sounds like a good idea I can send a pull request.

**Comment by pveyes** [comment]
Yes, I agree that as this is a complete rewrite, using ES2015+ and Babel would be the next logical steps considering OOP-style codebase. For the test startup time concern, I used [AVA](https://github.com/sindresorhus/ava) before and it's faster than Mocha + Babel (with ES2015 support out of the box). Not sure if it worth the change as test startup time only affect library developers, not user

**Comment by luigiplr** [comment]
@ekmartin +1, Have you already started? if so mind throwing a link to the repo :)

**Comment by ekmartin** [comment]
Not yet @luigiplr, thought I'd wait and hear @l12s' opinion first.

**Comment by luigiplr** [comment]
@ekmartin Began a bit of a refactor, along with some ideas for some api changes see: #25

**Comment by l12s** [comment]
Sorry, distracted by holidays and catch-up following them.

Happy to talk about this further, but I'd like to get the v2.0.0 release out and make sure that it's usable and supports client needs before looking at ES6+

So, my feeling on this is it something that we should pick up and talk about in a couple months once the client release is settled down.
